### PR TITLE
[Filter] Framework auto detection without specifying the option  @open sesame 2/03 15:26 

### PR DESCRIPTION
**Pull request by gichan-jang**
Modify the tensor filter to automatically detect the framework without specifying the framework option.
 The default value of the framework is changed to "auto".
 
Related issue: #2702 

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_plugins/unittest_plugins.cc [hunk 1]
```diff
@@ -4342,6 +4342,109 @@ TEST (test_tensor_filter, framework_auto_ext_tflite_wrong_inputtype_n)
...
+  const gchar fw_name[] = "tensorflow-lite";
+  GET_MODEL_PATH ("mobilenet_v1_1.0_224_quant.tflite");
+
+  str_launch_line = g_strdup_printf (
+      "videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter name=tfilter model=%s ! tensor_sink",
+      test_model);
+  gstpipe = gst_parse_launch (str_launch_line, NULL);
+  g_free (str_launch_line);
+  EXPECT_TRUE (gstpipe != nullptr);
+  TEST_TENSOR_FILTER_AUTO_OPTION_P (gstpipe, fw_name);
```
Comment (jaeyun-jung) [inline tests/nnstreamer_plugins/unittest_plugins.cc:4361]: plz add g_free (test_model);

File: tests/nnstreamer_plugins/unittest_plugins.cc [hunk 2]
```diff
@@ -4362,6 +4465,27 @@ TEST (test_tensor_filter, framework_auto_ext_tflite_nnfw_04)
...
+  const gchar fw_name[] = "nnfw";
+  GET_MODEL_PATH ("mobilenet_v1_1.0_224_quant.tflite");
+
+  str_launch_line = g_strdup_printf (
+      "videotestsrc ! videoconvert ! videoscale ! videorate ! video/x-raw,format=RGB,width=224,height=224 ! tensor_converter ! tensor_filter name=tfilter model=%s ! tensor_sink",
+      test_model);
+  gstpipe = gst_parse_launch (str_launch_line, NULL);
+  g_free (str_launch_line);
+  EXPECT_TRUE (gstpipe != nullptr);
+  TEST_TENSOR_FILTER_AUTO_OPTION_P (gstpipe, fw_name);
```
Comment (jaeyun-jung) [inline tests/nnstreamer_plugins/unittest_plugins.cc:4486]: also add g_free (test_model);

File: tests/nnstreamer_plugins/unittest_plugins.cc (hunk 1 above)
Comment (gichan-jang) [reply tests/nnstreamer_plugins/unittest_plugins.cc:4366]: It Is is released in `TEST_TENSOR_FILTER_AUTO_OPTION_P` macro.
But It causes confusion by performing free resources in the macro.
Changed to release the resource here.

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**